module/VuFindApi/tests/unit-tests/src/VuFindTest/Controller/SearchApiControllerTest.php

```diff
@@ -0,0 +1,299 @@
+                    'id' => 'record.1111',
+                ],
+                'headers' => [
+                    ['test-field', '51'],
+                ],
+            ],
+            [
+                'code' => 401,
+                'content' => '{"status":"UNAUTHORIZED","statusMessage":"API key invalid"}',
+            ],
+        ];
+        yield 'test keys enabled and not provided' => [
+            $config,
+            [
+                'queryAndPost' => [
+                    'id' => 'record.1111',
+                ],
+            ],
+            [
+                'code' => 200,
+                'content' => '{"resultCount":1,"records":[{"id":"record.1111","title":"hai!"}],"status":"OK"}',
+            ],
+        ];
+        $config['API_Keys']['mode'] = DeveloperSettingsStatus::ENFORCED->value;
+        yield 'test keys enforced and provided' => [
+            $config,
+            [
+                'queryAndPost' => [
+                    'id' => 'record.1111',
+                ],
+                'headers' => [
+                    ['test-field', '999999'],
+                ],
+            ],
+            [
+                'code' => 200,
+                'content' => '{"resultCount":1,"records":[{"id":"record.1111","title":"hai!"}],"status":"OK"}',
+            ],
+        ];
+        yield 'test keys enforced and not provided' => [
+            $config,
+            [
+                'queryAndPost' => [
+                    'id' => 'record.1111',
+                ],
+            ],
+            [
+                'code' => 401,
+                'content' => '{"status":"UNAUTHORIZED","statusMessage":"API key missing or invalid"}',
+            ],
+        ];
+    }
+
+    /**
+     * Get an instance of a searchApiController
+     *
+     * @param array $config      Main config
+     * @param array $paramsArray Parameters
+     *
+     * @return MockObject&SearchApiController
+     */
+    protected function createController(
+        array $config = [],
+        array $paramsArray = [],
+    ): MockObject&SearchApiController {
+        $solrOptions = $this->createMock(Options::class);
+        $solrOptions->expects($this->any())->method('getAPISettings')->willReturn([]);
+        $optionsPluginManager = $this->createMock(SearchPluginManager::class);
+        $optionsPluginManager->expects($this->any())->method('get')->willReturn($solrOptions);
+        $apiKeyMode = DeveloperSettingsStatus::fromSetting($config['API_Keys']['mode'] ?? '');
+        $apiKeysEnabled = DeveloperSettingsStatus::settingEnabled($apiKeyMode->value);
+        $developerSettingsService = $this->createMock(DeveloperSettingsService::class);
+        $developerSettingsService->expects($this->any())->method('apiKeysEnabled')->willReturn($apiKeysEnabled);
+        $developerSettingsService->expects($this->any())->method('getApiKeyMode')->willReturnCallback(
+            function () use ($apiKeyMode) {
+                return $apiKeyMode;
+            }
```

Comment:
I think a more concise approach to this could be:


```suggestion
            fn () => $apiKeyMode
```